### ISSUE TITLE
List of animated widgets/elements which are not disabled by "Remove Animations"

### ISSUE THREAD

**Issue by laundmo** [deckerst/aves#1102]
Here is a list of animated elements i've found which are not affected by "Remove Animations" setting, ordered by how important i consider them (completely subjective):

Fair enough:
- [x] In Settings or Image Metadata, clicking a collapsible title (settings categories, types of metadata) displays a flashing animation which expands quickly from the touch point to the edges of the title.
- [x] Buttons in the toolbar when viewing a image flash when pressed, the heart button animates a red section around the heart
- [x] Clicking a tag flashes, removing a tag from filter/search collapses it
- [x] In all scrollable views, scrolling and letting go, letting it scroll to the top or bottom, flashes a gray animated half-bubble overlay to show the end has been reached.
- [x] When viewing a image, sliding up/down has various sliding/fading animations after letting go.
- [x] In the Albums list, when clicking on a album after scrolling down, the hamburger menu and title "Collection" slide in from the top
- [x] The blue overlay when selecting images fades in
- [x] The checkmark for selecting images is animated, growing larger for a moment
- [x] Selecting the first image, or pressing the back button while selecting, animates the grouping title moving
- [x] Clicking on the search icon fades/animates in the search view

Blocked by framework (as of flutter v3.22.3):
- [ ] Opening the sidebar with the hamburger button animates it sliding open (framework drawer uses hard-coded duration `_kBaseSettleDuration`)
- [ ] The Kebab menu (⋮) option "View" triggers a popup which fades in (framework dialogs uses hard-coded duration of 150 ms)

Won't fix:
- [ ] When viewing a image, sliding left/right has various sliding/fading animations after letting go. There are multiple potential solutions:
  - Immediately display the correct view without any animation after letting go (slide with my finger, letting go makes it snap in place)
  - Don't slide the image at all, instead act on the sliding after letting go by immediately displaying the next image. This could display a icon on the corresponding screen edge when the slide has been recognized as such.
- [ ] On all gallery-like views, after scrolling down, the bar which allows switching to other views automatically slides up when scrolling up again
- [ ] In Map view, clicking on the image on the map will animate a small spinner around the edge of the preview below the map

Original post, not correct as per comments:

~**Is your feature request related to a problem? Please describe.**~
~I generally dislike animations like slides, fades, pop-in, spinners etc. i find them distracting and always disable animations in android accessibility settings.~

~Others use this setting for performance reasons, to avoid the performance impact of rendering the animations.~

~**Describe the solution you'd like**~
~Ideally this app would correctly use the Accessibility setting for removing animations, and display without animations when it is selected.~

~**Describe alternatives you've considered**~
~Alternatively a in-app setting to disable animations could work as well.~

~If none of the above is plausible, a separate build of the app without animations.~

~**Additional context**~
~I believe the Remove Animations option sets the animation speed settings to 0, so if its possible to directly use the animation speed settings it would also allow for the developer options which edit these directly to apply, like slower/faster animations without removing them entirely.~

**Comment by deckerst** [comment]
Have you tried Settings > Accessibility > Remove animations? Is it not working for you?

**Comment by laundmo** [comment]
> I'm asking because for me using "system default" does respect the Android settings (you may need to relaunch the app though).

You're absolutely right, i had set the theme to Black before and didn't consider that might have a impact. i'll edit this issue appropriately.

**Comment by laundmo** [comment]
hmm actually, i think i'll reopen it as a bug report then, as i feel strongly all themes should respect this setting.

**Comment by deckerst** [comment]
The theme shouldn't matter. I was talking about setting "system default" for Aves "remove animation" setting.

**Comment by laundmo** [comment]
Ahh, i'm sorry i got quite confused there for a second.
I thought you meant the android accessibility setting, as i had somehow overlooked the one in Aves. My bad.
Still, whether i use the android setting and "system default" in aves, or manually set it to remove animations in aves, some animations still play, which is what confused me.
I will collect a list of animations in a new issue.

**Comment by deckerst** [comment]
You can list them here, it's fine. I already know of some animations that remain (like the side drawer sliding) but at the moment I cannot remove them because the framework doesn't allow me. That said, I'm still interested in your list! If ambiguous, please provide a screen recording as well.

**Comment by laundmo** [comment]
> You can list them here, it's fine. I already know of some animations that remain (like the side drawer sliding) but at the moment I cannot remove them because the framework doesn't allow me. That said, I'm still interested in your list! If ambiguous, please provide a screen recording as well.

I have edited my original post with my list, as that allows you to use the Task List feature if you choose to and presents a more immediately visible list to others reading this

**Comment by deckerst** [comment]
Thanks. For some I disagree that the animation should be removed, for some I can't at the moment, but for some it's valid and I'll take a look. I'll try and put your list in tiers later.

**Comment by deckerst** [comment]
> On all gallery-like views, after scrolling down, the bar which allows switching to other views automatically slides up when scrolling up again

Currently the top bar and bottom nav bar both scroll away when scrolling down the view, and both scroll back in sight when scrolling up. This scrolling is proportional do the user scroll gesture. What do you expect when disabling animations?

> When viewing a image, sliding up, down, left, or right, has various sliding/fading animations after letting go.

I won't change that, but you may be interested in alternative gestures:
- instead of sliding left/right, enable `Settings` > `Viewer` > `Tap on screen edges to show previous/next item`, and then just tap on the sides of the viewer to go to the previous/next item instantly,
- instead of sliding down, just tap on the `back` button (if you use the old system navigation bar) or use the back gesture (if you use the new navigation gestures) to leave the viewer,
- instead of sliding up, use the `Info` action from the menu (or set it as a quick action for faster access), to go to the `Info` page.

**Comment by laundmo** [comment]
thank you!
